libs/store/file-system-tree.store.ts

```diff
@@ -0,0 +1,67 @@
+import { FileSystemTreeItemPresentationModel } from '@umbraco-cms/backoffice/backend-api';
+import { ArrayState, partialUpdateFrozenArray } from '@umbraco-cms/backoffice/observable-api';
+import { UmbStoreBase, UmbTreeStore } from '@umbraco-cms/backoffice/store';
+
+/**
+ * @export
+ * @class UmbFileSystemTreeStore
+ * @extends {UmbStoreBase}
+ * @description - General Tree Data Store
+ */
+export class UmbFileSystemTreeStore extends UmbStoreBase implements UmbTreeStore<FileSystemTreeItemPresentationModel> {
+	#data = new ArrayState<FileSystemTreeItemPresentationModel>([], (x) => x.path);
+
+	/**
+	 * Appends items to the store
+	 * @param {Array<FileSystemTreeItemPresentationModel>} items
+	 * @memberof UmbFileSystemTreeStore
+	 */
+	appendItems(items: Array<FileSystemTreeItemPresentationModel>) {
+		this.#data.append(items);
+	}
+
+	/**
+	 * Updates an item in the store
+	 * @param {string} path
+	 * @param {Partial<FileSystemTreeItemPresentationModel>} data
+	 * @memberof UmbFileSystemTreeStore
+	 */
+	updateItem(path: string, data: Partial<FileSystemTreeItemPresentationModel>) {
+		this.#data.next(partialUpdateFrozenArray(this.#data.getValue(), data, (entry) => entry.path === path));
```

Comment:
Here you can use this method instead, that would do the same:

```suggestion
		this.#data.appendOne(data)
```